packages/front-end/components/Experiment/MetricsOverridesSelector.tsx

```diff
@@ -52,20 +53,20 @@ export default function MetricsOverridesSelector({
         const metricDefinition = metricDefinitions.find(
           (md) => md.id === mo.id
         );
-        if (!metricDefinition) return;
 
-        // @ts-expect-error TS(2345) If you come across this, please fix it!: Argument of type 'number | undefined' is not assig... Remove this comment to see the full error message
-        const loseRisk = isNaN(mo.loseRisk)
-          ? metricDefinition.loseRisk
-          : // @ts-expect-error TS(2532) If you come across this, please fix it!: Object is possibly 'undefined'.
-            mo.loseRisk / 100;
-        // @ts-expect-error TS(2345) If you come across this, please fix it!: Argument of type 'number | undefined' is not assig... Remove this comment to see the full error message
-        const winRisk = isNaN(mo.winRisk)
-          ? metricDefinition.winRisk
-          : // @ts-expect-error TS(2532) If you come across this, please fix it!: Object is possibly 'undefined'.
-            mo.winRisk / 100;
-        // @ts-expect-error TS(2532) If you come across this, please fix it!: Object is possibly 'undefined'.
-        if (loseRisk < winRisk) {
+        const loseRisk =
+          isUndefined(mo.loseRisk) || isNaN(mo.loseRisk)
+            ? metricDefinition?.loseRisk
+            : mo.loseRisk / 100;
+        const winRisk =
+          isUndefined(mo.winRisk) || isNaN(mo.winRisk)
+            ? metricDefinition?.winRisk
+            : mo.winRisk / 100;
+        if (
+          !isUndefined(loseRisk) &&
+          !isUndefined(winRisk) &&
+          loseRisk < winRisk
+        ) {
```

Comment:
This changes the previous logic I had. Did you test the following scenario:

- Archive a metric
- Visit an experiment that uses that archived metric
- Open the edit metrics modal for the provided experiment

Reply:
`main` actually currently crashes for me when going through the flow above. My branch does not crash, but the view is not ideal.

Really, we should be removing archived metrics from experiment pages and related modals altogether, not just making the typing safe. And regenerating an experiment snapshot should probably exclude currently-archived metrics. But let's leave this for another task since it could get messy.

Reply:
Chatted on Slack, the remaining crash is specific to an archived denominator metric.